src/render-instruction.ts

```diff
@@ -116,13 +117,38 @@ ${typeMapperImports.join('\n')}`.trim()
   // Accounts
   // -----------------
   private processIxAccounts(): ProcessedAccountKey[] {
-    return this.ix.accounts.map((acc) => {
-      const knownPubkey = resolveKnownPubkey(acc.name)
-      const optional = acc.optional ?? false
-      return knownPubkey == null
-        ? { ...acc, optional }
-        : { ...acc, knownPubkey, optional }
+    let processedAccountsKey: ProcessedAccountKey[] = []
+    this.ix.accounts.map((acc: IdlInstructionAccount | IdlAccountsCollection) => {
+      if(this.isAccountsCollection(acc)){
+        acc.accounts.map((ac) => {
+          // as there are cases where the collection of the accounts is reused on the same ix, is needed to change the name of the account
+          ac.name += acc.name.charAt(0).toUpperCase().concat(acc.name.slice(1))
```

Comment:
I'm not a huge fan of this, what if there is an account that has the same name but with uppercase, i.e. `Accounts`?
Seems to be somewhat of a hack to me and I don't understand fully why the name as to be changed.

It seems to me the tool generating the IDL should choose a name instead that is less likely to clash, i.e.`__accounts__`.

At any rate all the data read from the IDL should be considered immutable as it is processed in other places as well and it is super hard to track down issues if it is mutated in multiple places.
If you look through the code base (aside from fixing some anchor issues) we never mutate any data read from the IDL, but derive other data from it in a readonly fashion instead.